### ISSUE TITLE
Weblate explodes when source string has special characters

### ISSUE THREAD

**Issue by albertvaka** [WeblateOrg/weblate#1702]
Weblate explodes with `invalid regular expression: invalid escape \ sequence` when trying to open a string to translate from Spanish imported from XLIFF that has accents on the source string (eg: "Cómo funciona").

Relevant part of the stacktrace:
```
/usr/local/lib/python2.7/dist-packages/weblate/trans/views/edit.py in translate
            'addword_form': InlineWordForm(), ...
/usr/local/lib/python2.7/dist-packages/weblate/trans/util.py in render
    return django_render(request, template, context, status=status) 
```

I've tested it with the latest version from master (from a week ago) and doesn't seem to be fixed.

**Comment by nijel** [comment]
Can you please post full backtrace? Also can you share the problematic file?

**Comment by albertvaka** [comment]
Here they go: [weblate_crash.zip](https://github.com/WeblateOrg/weblate/files/1532654/weblate_crash.zip)

**Comment by albertvaka** [comment]
```

Request Method: GET
Request URL: https://weblate.intranet/translate/verse-webapp/verse-web/es/?type=all

Django Version: 1.11.7
Python Version: 2.7.13
Installed Applications:
(u'django.contrib.auth',
 u'django.contrib.contenttypes',
 u'django.contrib.sessions',
 u'django.contrib.sites',
 u'django.contrib.messages',
 u'django.contrib.staticfiles',
 u'django.contrib.admin.apps.SimpleAdminConfig',
 u'django.contrib.admindocs',
 u'django.contrib.sitemaps',
 u'social_django',
 u'crispy_forms',
 u'compressor',
 u'rest_framework',
 u'rest_framework.authtoken',
 u'weblate.trans',
 u'weblate.lang',
 u'weblate.permissions',
 u'weblate.screenshots',
 u'weblate.accounts',
 u'weblate.utils',
 u'weblate.wladmin',
 u'weblate',
 u'weblate.gitexport')
Installed Middleware:
(u'django.contrib.sessions.middleware.SessionMiddleware',
 u'django.middleware.common.CommonMiddleware',
 u'django.middleware.locale.LocaleMiddleware',
 u'django.middleware.csrf.CsrfViewMiddleware',
 u'weblate.accounts.middleware.AuthenticationMiddleware',
 u'django.contrib.messages.middleware.MessageMiddleware',
 u'django.middleware.clickjacking.XFrameOptionsMiddleware',
 u'social_django.middleware.SocialAuthExceptionMiddleware',
 u'weblate.accounts.middleware.RequireLoginMiddleware',
 u'weblate.middleware.SecurityMiddleware')


Template error:
In template /usr/local/lib/python2.7/dist-packages/weblate/templates/glossary-embed.html, error at line 8
   invalid regular expression: invalid escape \ sequence
   1 : {% load i18n %}
   2 : {% load permissions %}
   3 : {% load translations %}
   4 : 
   5 : {% can_translate user unit.translation as user_can_translate %}
   6 : {% can_suggest user unit.translation as user_can_suggest %}
   7 : 
   8 :  {% for item in glossary %} 
   9 :     <tr>
   10 :     <td>{{ item.source }}</td>
   11 :     <td class="target">{% format_translation item.target unit.translation.language simple=True %}</td>
   12 :     <td>
   13 :     {% if user_can_translate or user_can_suggest %}
   14 :     <a class="copydict btn btn-default btn-xs" title="{% trans "Copy word to translation" %}">{% trans "Copy" %}</a>
   15 :     {% endif %}
   16 :     </td>
   17 :     </tr>
   18 : {% empty %}


Traceback:

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in _legacy_get_response
  249.             response = self._get_response(request)

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/usr/local/lib/python2.7/dist-packages/weblate/trans/views/edit.py" in translate
  605.             'addword_form': InlineWordForm(),

File "/usr/local/lib/python2.7/dist-packages/weblate/trans/util.py" in render
  208.     return django_render(request, template, context, status=status)

File "/usr/local/lib/python2.7/dist-packages/django/shortcuts.py" in render
  30.     content = loader.render_to_string(template_name, context, request, using=using)

File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py" in render_to_string
  68.     return template.render(context, request)

File "/usr/local/lib/python2.7/dist-packages/django/template/backends/django.py" in render
  66.             return self.template.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  207.                     return self._render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py" in render
  177.             return compiled_parent._render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py" in render
  72.                 result = block.nodelist.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/defaulttags.py" in render
  529.             return self.nodelist.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py" in render
  216.                 return template.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  209.                 return self._render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/usr/local/lib/python2.7/dist-packages/django/template/defaulttags.py" in render
  173.             len_values = len(values)

File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py" in __len__
  232.         self._fetch_all()

File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py" in _fetch_all
  1118.             self._result_cache = list(self._iterable_class(self))

File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py" in __iter__
  53.         results = compiler.execute_sql(chunked_fetch=self.chunked_fetch)

File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py" in execute_sql
  894.             raise original_exception

Exception Type: DataError at /translate/verse-webapp/verse-web/es/
Exception Value: invalid regular expression: invalid escape \ sequence
```

**Comment by nijel** [comment]
Duplicate of #1666 

Sorry for not recognizing the error earlier, but this issue has already been fixed in 7425a74
